src/pim-types.ts

```diff
@@ -255,6 +264,57 @@ export class TaskType extends EventType {
     ret.color = this.color;
     return ret;
   }
+
+  public getNextOccurence(): TaskType | null {
+    if (!this.isRecurring()) {
+      return null;
+    }
+
+    const rrule = this.rrule.clone();
+
+    if (rrule.count && rrule.count <= 1) {
+      return null; // end of reccurence
+    }
+
+    rrule.count = null; // clear count so we can iterate as many times as needed
+    const recur = rrule.iterator(this.startDate ?? this.dueDate);
+    let next = recur.next();
```

Comment:
Also, maybe consider renaming this variable, because it's a bit non-obvious. I'm sorry for the nit-picking here, it's just something I keep on coming back to and getting confused by when reviewing this.

Reply:
Renaming `next`? Would `nextRecurrence` be better?

Reply:
Yeah, that sounds better.